worker/restore_map.go

```diff
@@ -342,11 +351,24 @@ func (m *mapper) processReqCh(ctx context.Context) error {
 			return nil
 		}
 
+		if isNameSpaceAwareRestore {
+			// update a key to default namespace if user has requested namespace aware restore
+			// i.e 1-key => 0-key
+			parsedKey.Attr = strings.Replace(parsedKey.Attr, strconv.FormatUint(ns, 16), "0", 1)
```

Comment:
we should not use strings.Replace. Predicates could have a the namespace number too. See if we could add a test with such a predicate.